packages/zoe/test/unitTests/contracts/escrowToVote.js

```diff
@@ -0,0 +1,118 @@
+/* global harden */
+// @ts-check
+
+// Eventually will be importable from '@agoric/zoe-contract-support'
+import { assert } from '@agoric/assert';
+import makeStore from '@agoric/store';
+import { makeZoeHelpers } from '../../../src/contractSupport';
+
+/**
+ * Implement coin voting. Give a voting capability when a payment is
+ * escrowed.
+ *
+ * @typedef {import('../../../src/zoe').ContractFacet} ContractFacet
+ * @typedef {import('@agoric/ERTP').Amount} Amount
+ * @param {ContractFacet} zcf
+ */
+const makeContract = zcf => {
+  const { assertKeywords, checkHook } = makeZoeHelpers(zcf);
+  assertKeywords(harden(['Assets']));
+  const {
+    terms: { question },
+  } = zcf.getInstanceRecord();
+
+  const offerHandleToResponse = makeStore('offerHandle');
+
+  // `question` is a string defined in the terms.
+  // We assume the only valid answers are 'YES' and 'NO'
+  const validateResponse = response => {
+    assert(
+      Object.keys(response).length === 1 &&
+        Object.keys(response)[0] === question,
+      `The question '${
+        Object.keys(response)[0]
+      }' did not match the question '${question}'`,
+    );
+
+    assert(
+      response[question] === 'NO' || response[question] === 'YES',
+      `the answer '${response[question]}' was not 'YES' or 'NO'`,
+    );
+    // Throw an error if the response is not valid, but do not
+    // complete the offer. We should allow the voter to recast their vote.
+  };
+
+  const voterHook = voterOfferHandle => {
+    const voter = harden({
+      /**
+       * Vote on a particular issue
+       * @param {object} response: { [question]: answer }
+       */
+      vote: response => {
+        // Throw if the offer is no longer active, i.e. the user has
+        // completed their offer and the assets are no longer escrowed.
+        assert(
+          zcf.isOfferActive(voterOfferHandle),
+          `the escrowing offer is no longer active`,
```

Comment:
For example, this one has no substitution value and could just as easily be a string literal rather than a plain template literal. Were it a `details` template literal, if someone adds a substitution value, they will not accidentally disclose too much info. As a string literal, if someone adds a substitution value, it won't work until they change it to some kind of template literal. As a plain template literal you have the worst of both worlds.

Reply:
Ok, I will change that